rumqttd/src/router/shared_subs.rs

```diff
@@ -0,0 +1,129 @@
+use rand::Rng;
+use serde::{Deserialize, Serialize};
+
+pub struct SharedGroup {
+    // using Vec over HashSet for maintaining order of iter
+    clients: Vec<String>,
```

Comment:
Maybe I am remembering wrong but didn't you say we could move this to use the ClientId idx? Would be more nice than having Strings. (Although, if we ever do hash thing we might have to move it back to String right?)

Reply:
I thought `Connection` struct has connection id, but it doesn't, it also had `client_id` as String 😅 can't find a way to get connection id ( which is integer ) in forward_device_data.

Reply:
> Although, if we ever do hash thing we might have to move it back to String right?)

actually, how emqx does it is they will calculate phash and then mod it with len to get the client index ( plz correct me if wrong ). I can't see the point where / why it is useful.